src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,94 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.exception.DukeException;
+import duke.ui.UserInputType;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+
+    /**
+     * Parses String from user input into a LocalDateTime object.
+     *
+     * @param dateTimeString String input entered by user.
+     * @return LocalDateTime object of a task.
+     */
+    public static LocalDateTime parseDateTimeFromInput(String dateTimeString) {
+        DateTimeFormatter inputDateTimeFormatter = DateTimeFormatter.ofPattern("d/MM/yyyy Hmm");
+        return LocalDateTime.parse(dateTimeString, inputDateTimeFormatter);
+    }
+
+    /**
+     * Parses String from file into a LocalDateTime object.
+     *
+     * @param dateTimeString String from file.
+     * @return LocalDateTime object of a task.
+     */
+    public static LocalDateTime parseDateTimeFromFile(String dateTimeString) {
+        DateTimeFormatter inputDateTimeFormatter = DateTimeFormatter.ofPattern("MMM d yyyy hh:mm a");
+        return LocalDateTime.parse(dateTimeString, inputDateTimeFormatter);
+    }
+
+    /**
+     * Parses LocalDateTime object into a formatted string.
+     *
+     * @param dateTime LocalDateTime object for parsing.
+     * @return Formatted string representing the time of a task.
+     */
+    public static String parseDateTimeToString(LocalDateTime dateTime) {
+        return dateTime.format(DateTimeFormatter.ofPattern("MMM d yyyy hh:mm a"));
+    }
+
+    /**
+     * Parses user input into command for execution.
+     *
+     * @param userInput Input entered by user.
+     * @return Command object based on user input.
+     * @throws DukeException Unrecognized input by user.
+     */
+    public static Command parse(String userInput) throws DukeException {
+        String[] userInputArr = userInput.split(" ", 2);
+        if (userInputArr.length != 2) {
+            userInputArr = new String[]{userInputArr[0], ""};
+        }
+        String details = userInputArr[1].trim();
+
+        switch (parseUserInputType(userInputArr[0].trim())) {
+            case TODO:
+                return new AddTodoCommand(details);
+            case DEADLINE:
+                return new AddDeadlineCommand(details);
+            case EVENT:
+                return new AddEventCommand(details);
+            case DONE:
+                return new MarkTaskAsDoneCommand(details);
+            case FIND:
+                return new FindTaskCommand(details);
+            case DELETE:
+                return new DeleteTaskCommand(details);
+            case LIST:
+                return new ListTaskCommand(details);
+            case BYE:
+                return new ByeCommand(details);
+            default:
+                throw new DukeException("Sorry, I dont understand what that means :-(");
+        }
+    }
```

Comment:
switch statements: should each case statement be at the same indentation as switch (i.e no tab before case)?

Reply:
I agree with @jrvslam in that there should be no indentation before the case statements.

If IntelliJ is your IDE, you can follow the instructions at the following link for how to get IntelliJ to follow this coding convention for you.
https://stackoverflow.com/questions/40888798/config-intellij-switch-case-style#:~:text=Go%20to%20Settings%20%3E%20Editor%20%3E%20Code,Indent%20'case'%20branches%22.

Reply:
@jrvslam @douglaswja Thank you for pointing this out and the additional tip on IntelliJ setting! I agree with both of you and will make the changes accordingly!